### PR TITLE
Replaces BulkFailureIT with new test

### PR DESCRIPTION
**Pull request by keith-turner**
BulkFailureIT excercised low level bulk import implementation details that no longer exists in the elasticity branch.  The test was removed and a new test was added that operates at a much higher level.  The new test runs concurrent bulk import and compactions in such a way that duplicate loads of a file could be detected.

### CODE REVIEW

File: test/src/main/java/org/apache/accumulo/test/functional/BulkNewIT.java [hunk 1]
```diff
@@ -621,6 +623,65 @@ public void testManyFiles() throws Exception {
...
+
+      // Do N bulk imports of the exact same data.
+      for (int i = 0; i < N; i++) {
+        // Create 10 files for the bulk import.
+        for (int f = 0; f < 10; f++) {
+          writeData(dir + "/f" + f + ".", aconf, f * 1000, (f + 1) * 1000 - 1);
+        }
+        c.tableOperations().importDirectory(dir).to(tableName).tableTime(true).load();
+        getCluster().getFileSystem().delete(new Path(dir), true);
+      }
```
[the remark this answers is not in the document]
Comment (keith-turner) [reply test/src/main/java/org/apache/accumulo/test/functional/BulkNewIT.java:655]: > Not sure if its a good idea or not but could do something like this to kick off the bulk imports in parallel.

Its an interesting idea.  Running the imports serially vs in parallel each have the opportunity to expose different bugs. Going to experiment with doing both in two test to see what that looks like for code sharing.

[the remark this answers is not in the document]
Comment (keith-turner) [reply test/src/main/java/org/apache/accumulo/test/functional/BulkNewIT.java]: @DomGarguilo I updated the test in 2798d19 to run the bulk imports in parallel and serially.  I did not use the `parallel()` stream method as it uses a JVM wide thread pool and I am uncertain about the implications of placing task that do I/O on that thread pool.

[the remark this answers is not in the document]
Comment (DomGarguilo) [reply test/src/main/java/org/apache/accumulo/test/functional/BulkNewIT.java]: Makes sense to me. The code you added for this looks good to me.

**Review by DomGarguilo — APPROVED** [review]

**Review by EdColeman — APPROVED** [review]

**Comment by dlmarion** [comment]
This PR fixes #3611

**Pull request merged**